src/keys/ed25519-class.js

```diff
@@ -84,8 +83,8 @@ class Ed25519PrivateKey {
    * @returns {Promise<String>}
    */
   async id () {
-    const hash = await this.public.hash()
```

Comment:
Should the `hash` function on the public key return the identity hash? The public and private keys still have sha2-256 hash functions associated with them.

cc @jacobheun @vasco-santos

Reply:
@aschmahmann For what it's worth, the peer ID generated here matches those generated for signed IPNS records, and so this appears to follow the correct spec.

Reply:
Sure, you've made the `id` function behave correctly. However, I'm not sure what the `public.hash()` function is used for and that is potentially another bug.

Reply:
`public.hash()` calls `sha.multihashing(this.bytes, 'sha2-256')`, effectively returning a SHA2-based multihash of the public key. I used this call specifically because it followed the spec, and because the same call was used elsewhere in functions of the same depth in the same prototype.

Reply:
The hash should likely just be returning the identity hash, but that would be breaking across the api so that should happen outside of this, perhaps when we revisit reducing the bundle size of this repo for browsers.

This change looks good and fixes the existing bug.